batch2/batch/worker.py

```diff
@@ -504,31 +518,53 @@ def id(self):
     async def get_log(self):
         return {name: await c.get_log() for name, c in self.containers.items()}
 
-    async def delete(self):
+    async def delete(self, worker):
         log.info(f'deleting {self}')
         self.deleted = True
         for _, c in self.containers.items():
             await c.delete()
 
+        if self.start_time and not self.end_time:
+            await self.mark_cancelled(worker, time.time())
+
+    async def mark_started(self, worker, start_time):
+        assert not self.start_time and not self.end_time
+        self.start_time = start_time
+        asyncio.ensure_future(worker.post_job_timing(self))
+
+    async def mark_cancelled(self, worker, end_time):
+        self.mark_ended(end_time)
+        asyncio.ensure_future(worker.post_job_timing(self))
+
+    def mark_ended(self, end_time):
+        assert self.start_time and not self.end_time
+        self.end_time = end_time
```

Comment:
It is a little surprising this one doesn't post the job timings, but the other mark_ functions do.